### PR TITLE
[IOTDB-1745] Metagroup snapshots support template

### PR DESCRIPTION
**Pull request by stormbroken**
Snapshots of the meta group in the current cluster module support storage groups and TTL information, etc, but do not support templates. Templates need to be supported so that newly added nodes can register templates during cluster expansion.

I add test in cluster/log/snapshot/MetaSimpleSnapshotTest(testSerialize and testInstall) and cluster/server/member/MetaGroupMemberTest(testSendSnapshot)

### CODE REVIEW

File: cluster/src/test/java/org/apache/iotdb/cluster/log/snapshot/MetaSimpleSnapshotTest.java [hunk 1]
```diff
@@ -44,8 +49,7 @@
 import java.util.HashMap;
 import java.util.Map;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
```
Comment (neuyilan) [inline cluster/src/test/java/org/apache/iotdb/cluster/log/snapshot/MetaSimpleSnapshotTest.java:52]: In general, we don't need to reference packages using .*

**Review by neuyilan — APPROVED** [review]
LGTM, only minor issue.

File: cluster/src/test/java/org/apache/iotdb/cluster/log/snapshot/MetaSimpleSnapshotTest.java (hunk 1 above)
Comment (stormbroken) [reply cluster/src/test/java/org/apache/iotdb/cluster/log/snapshot/MetaSimpleSnapshotTest.java]: I fix this problems

File: cluster/src/main/java/org/apache/iotdb/cluster/log/snapshot/MetaSimpleSnapshot.java [hunk 2]
```diff
@@ -115,6 +125,13 @@ public ByteBuffer serialize() {
         dataOutputStream.write(entry.getValue().serialize().array());
       }
 
+      dataOutputStream.writeInt(templateMap.size());
+      for (Map.Entry<String, Template> entry : templateMap.entrySet()) {
+        SerializeUtils.serialize(entry.getKey(), dataOutputStream);
+        logger.info("A template into snapshot: {}", entry.getValue());
```
[the remark this answers is not in the document]
Comment (stormbroken) [reply cluster/src/main/java/org/apache/iotdb/cluster/log/snapshot/MetaSimpleSnapshot.java]: I fix them all

**Pull request merged**